Added Elective Lesson on Angular

HarshCasper opened the pull request:
Closes issue #1227 

@vegetabill Please review my Pull Request. I have added the necessary links and resources on Angular. If any further changes are needed, kindly let me know. Thanks.

Kundan28 commented:
> Closes issue #1227

Try writing `Fixed issue #1227 `

Kundan28 commented on Angular/angular.md:
```diff
@@ -0,0 +1,51 @@
+
+Angular is a popular Javascript Framework that is being used to build Applications for Web, Desktop and Mobile Devices. Launched by Google in 2010 as AngularJS, the subsequent versions of AngularJS are simply called Angular and involves the usage of Typescript which is a superset of Javascript. Typescript is compatible with Javascript and compiles to Javascript which is rendered on the Browser.
+
+Angular is similar to React.js and Vue which are being used for creating Front-End User Interfaces. What makes Angular quite intutive and easy to learn is that, Angular uses HTML to define the User-Interface and hence we don't need to write the Front-End Interface in Javascript. HTML is a Markup language and follows a well-defined layout where we can bind the data using Angular and readily render ...
```
Better to add these as a bullet points.

Kundan28 commented on Angular/angular.md:
```diff
@@ -0,0 +1,51 @@
+Angular is a popular Javascript Framework that is being used to build Applications for Web, Desktop and Mobile Devices. Launched by Google in 2010 as AngularJS, the subsequent versions of AngularJS are simply called Angular and involves the usage of Typescript which is a superset of Javascript. Typescript is compatible with Javascript and compiles to Javascript which is rendered on the Browser.
+
+Angular is similar to React.js and Vue which are being used for creating Front-End User Interfaces. What makes Angular quite intutive and easy to learn is that, Angular uses HTML to define the User-Interface and hence we don't need to write the Front-End Interface in Javascript. HTML is a Markup language and follows a well-defined layout where we can bind the data using Angular and readily render ...
+## What problems does it solve?
```
Add a line space before this ?

Kundan28 commented on Angular/angular.md:
```diff
@@ -0,0 +1,51 @@
+
+Angular comes built-in with multiple features like a Template Engine, Form Validation, CLI Tool and finally the RXJS which is used to make HTTP Request from an Angular Application. 
+
+
```
Remove extra line spaces.

HarshCasper commented:
I have updated the changes @Kundan28 

Kindly review it once and let me know if further changes are needed. If it is okay, kindly accept this Pull Request.

Kundan28 commented on Angular/angular.md:
```diff
@@ -3,30 +3,26 @@
 ## Motivation
 
-Angular is a popular Javascript Framework that is being used to build Applications for Web, Desktop and Mobile Devices. Launched by Google in 2010 as AngularJS, the subsequent versions of AngularJS are simply called Angular and involves the usage of Typescript which is a superset of Javascript. Typescript is compatible with Javascript and compiles to Javascript which is rendered on the Browser.
+- Angular is a popular Javascript Framework that is being used to build Applications for Web, Desktop and Mobile Devices. Launched by Google in 2010 as AngularJS, the subsequent versions of AngularJS are simply called Angular and involves the usage of Typescript which is a superset of Javascript. Typescript is compatible with Javascript and compiles to Javascript which is rendered on the Browser.
```
I meant to say that convert this paragraph into small bullet points, not the entire paragraph to single bullet point.

Kundan28 commented on Angular/angular.md:
```diff
@@ -3,30 +3,26 @@
+- Angular is a popular Javascript Framework that is being used to build Applications for Web, Desktop and Mobile Devices. Launched by Google in 2010 as AngularJS, the subsequent versions of AngularJS are simply called Angular and involves the usage of Typescript which is a superset of Javascript. Typescript is compatible with Javascript and compiles to Javascript which is rendered on the Browser.
 
-Angular is similar to React.js and Vue which are being used for creating Front-End User Interfaces. What makes Angular quite intutive and easy to learn is that, Angular uses HTML to define the User-Interface and hence we don't need to write the Front-End Interface in Javascript. HTML is a Markup language and follows a well-defined layout where we can bind the data using Angular and readily render ...
+- Angular is similar to React.js and Vue which are being used for creating Front-End User Interfaces. What makes Angular quite intutive and easy to learn is that, Angular uses HTML to define the User-Interface and hence we don't need to write the Front-End Interface in Javascript. HTML is a Markup language and follows a well-defined layout where we can bind the data using Angular and readily rend ...
```
Same here.

Kundan28 commented on Angular/angular.md:
```diff
@@ -3,30 +3,26 @@
+- Angular is similar to React.js and Vue which are being used for creating Front-End User Interfaces. What makes Angular quite intutive and easy to learn is that, Angular uses HTML to define the User-Interface and hence we don't need to write the Front-End Interface in Javascript. HTML is a Markup language and follows a well-defined layout where we can bind the data using Angular and readily rend ...
 ## What problems does it solve?
 One of the greatest advantages with Angular is that, it is Cross-Platform and Modular. Angular can be used for developing Web Application along with Progressive Web Applications, Desktop Applications and Native Mobile Applications. Angular is also modular as the whole code is divided into "modules" which consists of components, pipes and directives coupled together. These modules are reusable and ...
-
```
No need to remove this.

Kundan28 commented on Angular/angular.md:
```diff
@@ -3,30 +3,26 @@
-
 ### Why is it recommended?
 Angular is recommended to learn because of the Model View Controller (MVC) Architecture that it follows which allows the developers to do more while written lesser lines of code. Angular is also pre-built in many existing frameworks like Ionic and Wejmo and hence allows to add attractive User-Interfaces. Angular also supports Single Page Applications which are updated dynamically by communicating ...
-
```
No need to remove this.

Kundan28 commented on Angular/angular.md:
```diff
@@ -3,30 +3,26 @@
 - [Github Community Forum](https://github.community/)
 - [Microsoft Office Home](https://www.office.com/apps?auth=2)
 - [Forbes](https://www.forbes.com/sites/karstenstrauss/2019/01/22/the-most-sustainable-companies-in-2019)
-
```
No need to remove this.

Kundan28 commented on Angular/angular.md:
```diff
@@ -3,30 +3,26 @@
 - [Forbes](https://www.forbes.com/sites/karstenstrauss/2019/01/22/the-most-sustainable-companies-in-2019)
-
 ## Description
-
```
No need to remove this.

HarshCasper commented:
I have updated the File as mentioned. I have removed all the whitespaces and simplified the paragraphs by converting them into bullet-points. Spaces have been completely removed except for the headings where one space have been given. Kindly notify me for any further changes needed.

vegetabill commented on Angular/angular.md:
```diff
@@ -0,0 +1,45 @@
+- Typescript is compatible with Javascript and compiles to Javascript which is rendered on the Browser.
+- Angular is similar to React.js and Vue which are being used for creating Front-End User Interfaces.
+- What makes Angular quite intutive and easy to learn is that, Angular uses HTML to define the User-Interface and hence we don't need to write the Front-End Interface in Javascript. 
+- HTML is a Markup language and follows a well-defined layout where we can bind the data using Angular and readily render an interface with minimal effort.
```
Please condense this into a one sentence reason why someone who, say, already learned React should bother with Angular.

vegetabill commented on Angular/angular.md:
```diff
@@ -0,0 +1,45 @@
+One of the greatest advantages with Angular is that, it is Cross-Platform and Modular. Angular can be used for developing Web Application along with Progressive Web Applications, Desktop Applications and Native Mobile Applications. Angular is also modular as the whole code is divided into "modules" which consists of components, pipes and directives coupled together. These modules are reusable and ...
+
+### Why is it recommended?
+Angular is recommended to learn because of the Model View Controller (MVC) Architecture that it follows which allows the developers to do more while written lesser lines of code. Angular is also pre-built in many existing frameworks like Ionic and Wejmo and hence allows to add attractive User-Interfaces. Angular also supports Single Page Applications which are updated dynamically by communicating ...
```
Can you condense this entire section? It should be a few sentences, not a few paragraphs.

Most good frameworks are modular so this isn't really a reason to choose Angular. This might be difficult to do if you don't have a lot of expertise in multiple frameworks, so I'd suggest asking more experienced Angular fans why they prefer it.

vegetabill commented on Angular/angular.md:
```diff
@@ -0,0 +1,45 @@
+- [Google About](https://about.google/)
+- [Github Community Forum](https://github.community/)
+- [Microsoft Office Home](https://www.office.com/apps?auth=2)
+- [Forbes](https://www.forbes.com/sites/karstenstrauss/2019/01/22/the-most-sustainable-companies-in-2019)
```
These types of sections age too quickly. Please remove.

vegetabill commented on Angular/angular.md:
```diff
@@ -0,0 +1,45 @@
+Angular describes itself as "One framework. Mobile & desktop." It is a single-page Client Side Web Framework that follows the MVC Architecture to render Single Page Applications (SPA). Angular is a titan in the Cross-Platform Market alongside React and Flutter, and utilizes Document Object Model to render Web Interfaces. The major points with Angular is its robustness and Browser Support which ma ...
+- Angular utilizes Hypertext Markup Language (HTML), Cascading Style Sheets (CSS) and Typescript. Angular is an ever-growing technology and is increasingly being used by Developers to create Front-End Interface. 
+- Angular versions are backward-compatible which means that the latest version of the Angular will still work with the older versions. With the proposal of Ivy, which is a next-generation pipeline model to render Angular compilation, the runtime and load-time on slower networks will be enhanced.
+- Angular comes built-in with multiple features like a Template Engine, Form Validation, CLI Tool and finally the RXJS which is used to make HTTP Request from an Angular Application. 
```
I would condense this down to the first two sentences. The rest sounds like ad copy.

Sorry if the length or level of detail wasn't clear in the template. I've added more notes so it is more clear it should be short.

HarshCasper commented:
I have made subsequent changes. Kindly review the changes made and let me know if further changes are needed @vegetabill @Kundan28

HarshCasper commented:
I have made subsequent changes. Kindly review the changes made and let me know if further changes are needed @vegetabill @Kundan28

vegetabill commented:
Looks great @HarshCasper! Thanks for including this lesson.

Kundan28 commented:
Made the changes in the scoresheet too.
